src/mopidy/backend.py

```diff
@@ -472,7 +491,7 @@ class LibraryProviderProxy:
     browse = proxy_method(LibraryProvider.browse)
     get_distinct = proxy_method(LibraryProvider.get_distinct)
     get_images = proxy_method(LibraryProvider.get_images)
-    lookup = proxy_method(LibraryProvider.lookup)
+    lookup = proxy_method(LibraryProvider.wrapped_lookup)
```

Comment:
Hey @kingosticks sorry for the delay on this, but I took a stab at not breaking the API. There were some issues I ran into when trying to implement a non breaking change. One of the main ones is that because we're dealing with futures in the library caller, we can't know if the backend supports the new batching if we allow both batching and non batching until we evaluate the future. Since there can be multiple backends, we need to assume either they all support batching or none do as far as I could tell. This wrapper lets us assume they're all batched, and if the batched method isn't support, we just call the old lookup sequentially. 

I opted for a new method instead of updating the signature mostly because I felt it was cleaner. I think we could add a `uris: list[Uri] | None = None` optional param, but then we'd need to introspect the method signature in that wrapper or something, and it felt messy to me. I don't feel strongly on this, so happy to take a stab at a different idea too. 

The other thing was I wasn't sure how to test the mixing of non-batch supported backends and batch-supported backends. Since I opted to abstract this away in the proxy and treat the dummy backend as batched. To that point, it might just be a better idea to call the wrapper directly with this approach instead of hiding it in the proxy, but I figure in future releases when the single lookup gets deprecated, the wrapper can be deleted and the proxy updated to `lookup` again.

Reply:
Any thoughts on this approach @jodal @adamcik ? Another alternative might be to have a (temporary) `use_batched_lookup` property for supporting backends to define. But it is nicer to have the compatibility workaround stuff just in our code, rather than in backends (extensions) also.